### DISCUSSION TITLE
HLS latency

### DISCUSSION THREAD

**Discussion by Wheemer** [bluenviron/mediamtx#1087] [Questions]
Hello;

I am interested in the least delay possible and no buffering or seeking. Can you suggest good settings? HLS is always minutes behind rtsp. I am using this for security cameras and would like the most immediate image possible. I tried LL HLS but that just didn't work at all inside Emby my main client for the cameras.

Thanks!

**Comment by aler9** [comment]
Hello, RTSP is the protocol with the lowest latency, and i can assure you that latency added by the protocol is below 50ms. If you have latency issues with RTSP, it means that you should either tune the client (Emby) or find another client.

**Reply by Wheemer** [reply]
But RTSP links take longer to begin streaming in emby, that is why I chose HLS. However the HLS stream is usually 2 minutes behind the RTSP stream. I was looking to remove the buffering that causes the 2 minute delay.

**Reply by aler9** [reply]
RTSP delay is 50ms
HLS delay is 20-30 seconds.
Low-latency HLS delay is 2-3 seconds or even lower in some cases.

If the delay is greater, that is added by the client, in this case Emby.

[a comment by an automated account is not shown]
